src/components/dashboard/main-content/course-enrollments/CourseEnrollments.jsx

```diff
@@ -1,57 +1,62 @@
 import NewGroupAssignmentAlert from './NewGroupAssignmentAlert';
-
+import CourseEnrollmentsContext from './course-cards/CourseEnrollmentsContext';
+
+const messages = defineMessages({
+  inProgress: {
+    id: 'enterprise.dashboard.course.enrollment.moved.to.in.progress.alert.text',
+    defaultMessage: 'Your course was moved to "In Progress".',
+    description: 'Message when a course is moved to in progress.',
+  },
+  savedForLater: {
+    id: 'enterprise.dashboard.course.enrollment.saved.for.later.alert.text',
+    defaultMessage: 'Your course was saved for later.',
+    description: 'Message when a course is saved for later.',
+  },
+});
+
+/**
+ * Checks if the user has visited the dashboard before, determined by the presence of
+ * localStorage set by the `dashboardLoader`.
+ *
+ * @returns {boolean} - Whether the user has visited the dashboard before.
+ */
 function useIsFirstDashboardVisit() {
-  const [isFirstVisit, setIsFirstVisit] = useState(false);
-  useEffect(() => {
-    const cookies = new Cookies();
-    const hasUserVisitedDashboard = cookies.get('has-user-seen-enrollments');
-    if (!hasUserVisitedDashboard) {
-      cookies.set('has-user-seen-enrollments', true, { path: '/' });
-      setIsFirstVisit(true);
-    }
-  }, []);
-  return isFirstVisit;
+  const hasUserVisited = localStorage.getItem('has-user-visited-learner-dashboard');
+  return !hasUserVisited;
 }
 
-function useSaveForLaterAlerts() {
-  const { state } = useLocation();
-  const [shouldShowMarkSavedForLaterCourseSuccess, setShouldShowMarkSavedForLaterCourseSuccess] = useState(false);
-  const [shouldShowMoveToInProgressCourseSuccess, setShouldShowMoveToInProgressCourseSuccess] = useState(false);
-
-  useEffect(() => {
-    if (state?.markedSavedForLaterSuccess) {
-      setShouldShowMarkSavedForLaterCourseSuccess(true);
-    } else {
-      setShouldShowMarkSavedForLaterCourseSuccess(false);
-    }
-  }, [state?.markedSavedForLaterSuccess]);
-
-  useEffect(() => {
-    if (state?.markedInProgressSuccess) {
-      setShouldShowMoveToInProgressCourseSuccess(true);
-    } else {
-      setShouldShowMoveToInProgressCourseSuccess(false);
-    }
-  }, [state?.markedInProgressSuccess]);
-
-  return {
-    shouldShowMarkSavedForLaterCourseSuccess,
-    setShouldShowMarkSavedForLaterCourseSuccess,
-    shouldShowMoveToInProgressCourseSuccess,
-    setShouldShowMoveToInProgressCourseSuccess,
-  };
+function useEnrollmentStatusChangeAlerts() {
+  const intl = useIntl();
+  const [courseEnrollmentStatusChanges, setCourseEnrollmentStatusChanges] = useState([]);
+  const addCourseEnrollmentStatusChangeAlert = useCallback((type) => {
+    const message = intl.formatMessage(messages[camelCase(type)]);
+    const uuid = uuidv4();
+
+    const courseEnrollmentChangeAlert = {
+      uuid,
+      message,
+      dismiss: () => {
+        setCourseEnrollmentStatusChanges((prevState) => prevState.filter(alert => alert.uuid !== uuid));
```

Comment:
[suggestion] It may be worth getting coverage on dismissal of these alerts. It looks like we do have at least two test cases related to moving courses between in-progress and saved for later that could be extended to include dismissal.